src/coordination/coordinator_client.cpp

```diff
@@ -140,5 +141,23 @@ auto CoordinatorClient::SendUnregisterReplicaRpc(std::string const &instance_nam
   return false;
 }
 
+auto CoordinatorClient::SendGetInstanceUUID() const
+    -> utils::BasicResult<GetInstanceUUIDError, std::optional<utils::UUID>> {
+  std::optional<utils::UUID> instance_uuid;
```

Comment:
This doesn't seem logical to me: BasicResult<Error, optional>
Why not BasicResult<Error, UUID>. What is the point of returning empty optional

Reply:
Aha, my idea was that replica can have it, but it could be set to nullopt, or we could get error from RPC, so either case we need to deal with that, does that make sense?

Reply:
Resolved on call